containers/src/impl/Kokkos_Functional_impl.hpp

```diff
@@ -89,6 +89,8 @@ uint32_t MurmurHash3_x86_32(const void* key, int len, uint32_t seed) {
       k1 = rotl32(k1, 15);
       k1 *= c2;
       h1 ^= k1;
+      break;
+    default: break;
```

Comment:
I would just have suppressed the diagnostic for that one.

Reply:
We can do
```diff
diff --git a/containers/src/impl/Kokkos_Functional_impl.hpp b/containers/src/impl/Kokkos_Functional_impl.hpp
index 76d57ab2a..43d66a4af 100644
--- a/containers/src/impl/Kokkos_Functional_impl.hpp
+++ b/containers/src/impl/Kokkos_Functional_impl.hpp
@@ -80,7 +80,7 @@ uint32_t MurmurHash3_x86_32(const void* key, int len, uint32_t seed) {
 
   uint32_t k1 = 0;
 
-  switch (len & 3) {
+  switch (len & 3) {  // NOLINT(bugprone-switch-missing-default-case)
     case 3: k1 ^= tail[2] << 16; [[fallthrough]];
     case 2: k1 ^= tail[1] << 8; [[fallthrough]];
     case 1:
@@ -89,8 +89,6 @@ uint32_t MurmurHash3_x86_32(const void* key, int len, uint32_t seed) {
       k1 = rotl32(k1, 15);
       k1 *= c2;
       h1 ^= k1;
-      break;
-    default: break;
   };
 
   //----------
```
instead but I don't have a strong opinion either way (except that the CI is already passing).

Reply:
That's what I meant.  Waiting to hear from other reviewers what they think.

Reply:
Hmm I think I would go with putting a default like @masterleinad did

Reply:
I agree with an empty break; it's not reachable and it's true that the diagnostic is wrong in this case but I'd rather minimize diagnostic suppressions in favor of keeping a default

Reply:
Note that
```C++
[...]
    break;
  default:
};
```
would give me
```
warning: label at end of compound statement is a C++23 extension [-Wc++23-extensions]
```